webapp/src/components/BuyManaPage/BuyManaForm/BuyManaForm.js

```diff
@@ -0,0 +1,226 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { Button, Form, Input, Icon } from 'semantic-ui-react'
+import TxStatus from 'components/TxStatus'
+import { preventDefault } from 'lib/utils'
+import { t, t_html } from 'modules/translation/utils'
+import {
+  fetchTransaction,
+  fetchManaRate,
+  fetchManaDefaultRate,
+  getSlippage
+} from './utils'
+
+import './BuyManaForm.css'
+
+const FETCH_PRICE_INTERVAL = 10 * 1000
+
+export default class BuyManaForm extends React.PureComponent {
+  static propTypes = {
+    address: PropTypes.string,
+    balance: PropTypes.number,
+    isTxIdle: PropTypes.bool,
+    onSubmit: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.interval = null
+    this.state = {
+      amount: '',
+      defaultRate: null,
+      rate: null,
+      loading: false
+    }
+  }
+
+  componentDidMount() {
+    this.fetchDefaultRate()
+    this.clearInterval()
+    this.interval = setInterval(
+      () => this.fetchDefaultRate(),
+      FETCH_PRICE_INTERVAL
+    )
```

Comment:
I don't see the value of bringing `rate`, `defaultRate` and `tx` to the global state, since they only concern this component and they are polled/fetched often, that means if we move them to redux we'd be dispatching actions every few seconds, triggering lots of `mapStateToProp`s and if not reselected correctly, lots of `componentShouldUpdate`s and `render`s downstream. That's why I kept them as internal state and class methods.

The only thing that I left as internal state but I do see the value of moving to redux is the Ether balance, that one I think it could be useful at app level.

Reply:
✅  done (bc25f3e)